delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/StoragePluginModule.scala

```diff
@@ -283,4 +291,77 @@ class StoragePluginModule(priority: Int) extends ModuleDef {
   many[PriorityRoute].add { (fileRoutes: FilesRoutes) =>
     PriorityRoute(priority, fileRoutes.routes, requiresStrictEntity = false)
   }
+
+  if (MigrationState.isRunning) {
+    // Storages
+    many[MigrationLog].add { (cfg: StoragePluginConfig, xas: Transactors, clock: Clock[UIO], crypto: Crypto) =>
+      MigrationLog.scoped[Iri, StorageState, StorageCommand, StorageEvent, StorageRejection](
+        Storages.definition(
+          cfg.storages.storageTypeConfig,
+          (_, _) => IO.terminate(new IllegalStateException("Storage command evaluation should not happen")),
+          UIO.terminate(new IllegalStateException("Storage command evaluation should not happen")),
+          crypto
+        )(clock),
+        e => e.id,
+        identity,
+        (e, _) => injectStorageDefaults(cfg.defaults)(e),
+        cfg.storages.eventLog,
+        xas
+      )
+    }
+
+    // Files
+    many[MigrationLog].add { (cfg: StoragePluginConfig, xas: Transactors, clock: Clock[UIO]) =>
+      MigrationLog.scoped[Iri, FileState, FileCommand, FileEvent, FileRejection](
+        Files.definition(clock),
+        e => e.id,
+        enrichJsonFileEvent,
+        injectFileStorageInfo,
+        cfg.files.eventLog,
+        xas
+      )
+    }
+
+  }
+}
+
+// TODO: This object contains migration helpers, and should be deleted when the migration module is removed
+object StoragePluginModule {
+
+  private def setStorageDefaults(name: Option[String], description: Option[String]): StorageValue => StorageValue = {
+    case disk: DiskStorageValue         => disk.copy(name = name, description = description)
+    case s3: S3StorageValue             => s3.copy(name = name, description = description)
+    case remote: RemoteDiskStorageValue => remote.copy(name = name, description = description)
+  }
+
+  def injectStorageDefaults(defaults: Defaults): StorageEvent => StorageEvent = {
+    case s @ StorageCreated(id, _, value, _, _, _, _) if id == storages.defaultStorageId =>
+      s.copy(value = setStorageDefaults(Some(defaults.name), Some(defaults.description))(value))
+    case s @ StorageUpdated(id, _, value, _, _, _, _) if id == storages.defaultStorageId =>
+      s.copy(value = setStorageDefaults(Some(defaults.name), Some(defaults.description))(value))
+    case event                                                                           => event
+  }
+
+  def enrichJsonFileEvent: Json => Json = { input =>
```

Comment:
You should only inject for the events where the storage was not present before

Reply:
If the storage is present, this will not override it and keep the original storage (I added a test to make sure). But I will try to see how I can make this better

Reply:
Updated it to make it more explicit